### PR TITLE
feat(github): add go releaser

### PR DESCRIPTION
**Pull request by mazzy89**

### CODE REVIEW

**Comment by x1unix** [comment]
@mazzy89 can you also please add build for `GOOS=js GOARCH=wasm`?

Just for clarification - wouldn't it be an overkill to have a Go releaser for such a simple case?
Matrix strategy technically should be able to handle this.

**Comment by mazzy89** [comment]
Goreleaser handles already the creation of release etc... The Github action should have more jobs and hence be slower. While goreleaser handles everything internally. Further the repo had already gorelease manifest.

File: .goreleaser.yaml [hunk 1]
```diff
@@ -6,15 +6,17 @@ before:
...
+    binary: gnopls
     goos:
       - linux
       - darwin
+      - js
     goarch:
       - amd64
       - arm64
-    binary: gnopls
+      - wasm
```
Comment (x1unix) [inline .goreleaser.yaml:17]: Does it add `.wasm` extension to output file? This might be important for HTTP server to add a correct content type.

Comment (mazzy89) [reply .goreleaser.yaml:17]: Correct. It does.

**Pull request merged**